### ISSUE TITLE
Post subject with single quote won't show

### ISSUE THREAD

**Issue by MangaD** [MyBBStuff/MyAlerts#231]
Alerts will show an empty subject for subjects that contain a single quote.
In order to fix this I have changed the function below in the `/inc/plugins/MybbStuff/MyAlerts/src/AlertManager.php` file.

```
public function commit()
{
	if (empty(static::$alertQueue)) {
		$success = true;
	} else {
		$toCommit = array();

		foreach (static::$alertQueue as $alert) {
			$alertArray = $alert->toArray();

			// MangaD
			$alertArray['extra_details']['subject'] = $this->db->escape_string(
				$alertArray['extra_details']['subject']
			);

			$alertArray['extra_details'] = $this->db->escape_string(
				$alertArray['extra_details']
			);

			$toCommit[] = $alertArray;
		}

		$success = (boolean) $this->db->insert_query_multiple(
			'alerts',
			$toCommit
		);
	}

	return $success;
}
```

**Comment by MangaD** [comment]
That is interesting. Not only is it working as it fixed my problem...

**Comment by euantorano** [comment]
Weird. That code is also executed for every alert type and not every alert type has a `subject` extra detail, so I'd expect warnings about illegal offsets if nothing else...

**Comment by MangaD** [comment]
I have changed it to:
```
// MangaD
if (array_key_exists('subject', $alertArray['extra_details'])) {
	$alertArray['extra_details']['subject'] = $this->db->escape_string(
		$alertArray['extra_details']['subject']
	);
}
```
and it is still working.

**Comment by euantorano** [comment]
If you add a `var_dump($alertArray['extra_details'])` is the extra details key a string as it should be?

**Comment by MangaD** [comment]
Ok, so things are more complicated than I thought. `var_dump` still doesn't do anything even with `die()`, however, my problem was not fixed as I thought. Apparently the problem is not (just?) in the subject having a single quote, but when a user other than me quotes me on a post whose subject has a single quote. If I quote myself on that post then the subject shows right, but if another user does it then it shows wrong. If that same user mentions me on a post whose subject does not contain a single quote, then it works fine.

**Comment by euantorano** [comment]
Yeah, it’s a bit tangled in knots. I’ll have to do some debugging to see if I can work out what on Earth is going on.

**Comment by lairdshaw** [comment]
Hi @MangaD and @euantorano,

I'm working through the milestones for the next release, and this issue is on one of them. I can't, though, reproduce the bug. Did either of you ever get to the bottom of what this issue and/or its cause is, or at least come up with a recipe to replicate it?

**Comment by euantorano** [comment]
Hey,

I don't think I ever tracked it down, but it's been a while. There were similar reported issues for quote alerts sometimes not firing under similar circumstances too, but I never tracked those down either.

**Comment by lairdshaw** [comment]
> Hey,
> 
> I don't think I ever tracked it down, but it's been a while.

OK, thanks for letting me know.

> There were similar reported issues for quote alerts sometimes not firing under similar circumstances too, but I never tracked those down either.

Maybe fixed by #274?

**Comment by euantorano** [comment]
Yeah, possibly fixed by that but not completely sure. On 13 Mar 2024, at 09:02, Laird ***@***.***> wrote:﻿

Hey,
I don't think I ever tracked it down, but it's been a while.

OK, thanks for letting me know.

There were similar reported issues for quote alerts sometimes not firing under similar circumstances too, but I never tracked those down either.

Maybe fixed by #274?

**Comment by lairdshaw** [comment]
Closing in the absence of confirmation that this issue still exists. If it does, feel free to reopen it.
